cypress/integration/fileMenu.cy.js

```diff
@@ -0,0 +1,140 @@
+import { typeInput } from '../helpers/common.js'
+import { selectEventProgram } from '../helpers/dimensions.js'
+import { clickMenubarUpdateButton } from '../helpers/menubar.js'
+import { EXTENDED_TIMEOUT } from '../support/util.js'
+
+const ITEM_NEW = 'file-menu-new'
+const ITEM_OPEN = 'file-menu-open'
+const ITEM_SAVE = 'file-menu-save'
+const ITEM_SAVEAS = 'file-menu-saveas'
+const ITEM_RENAME = 'file-menu-rename'
+const ITEM_TRANSLATE = 'file-menu-translate'
+const ITEM_SHARING = 'file-menu-sharing'
+const ITEM_GETLINK = 'file-menu-getlink'
+const ITEM_DELETE = 'file-menu-delete'
+
+const defaultItemsMap = {
+    [ITEM_NEW]: true,
+    [ITEM_OPEN]: true,
+    [ITEM_SAVE]: false,
+    [ITEM_SAVEAS]: false,
+    [ITEM_RENAME]: false,
+    [ITEM_TRANSLATE]: false,
+    [ITEM_SHARING]: false,
+    [ITEM_GETLINK]: false,
+    [ITEM_DELETE]: false,
+}
+
+const assertItems = (enabledItemsMap = {}) => {
+    const itemsMap = Object.assign({}, defaultItemsMap, enabledItemsMap)
+
+    cy.getBySel('menubar').contains('File').click()
+
+    Object.keys(itemsMap).forEach((itemName) => {
+        itemsMap[itemName]
+            ? cy.getBySel(itemName).should('not.have.class', 'disabled')
+            : cy.getBySel(itemName).should('have.class', 'disabled')
+    })
+}
+
+describe('file menu', () => {
+    it('reflects empty state', () => {
+        cy.visit('/', EXTENDED_TIMEOUT)
+
+        assertItems()
+    })
+
+    it('reflects unsaved state', () => {
+        cy.visit('/', EXTENDED_TIMEOUT)
+
+        selectEventProgram({
+            programName: 'Adverse events following immunization',
+        })
+
+        clickMenubarUpdateButton()
+
+        assertItems({
+            [ITEM_SAVE]: true,
+        })
+    })
+
+    it('reflects saved state', () => {
+        cy.visit('/#/ZTrsv19jw9U', EXTENDED_TIMEOUT)
+
+        cy.getBySel('visualization-title').contains('COVAC enrollment')
+
+        assertItems({
```

Comment:
shouldn't `SAVE` be `true` here as well? If it's opening an AO where saving is allowed? (compared to the test below, where saving isn't allowed)

Reply:
No, SAVE is only allowed when it's `dirty`. Otherwise use SAVE AS.

Reply:
Fixed now